ros/src/computing/planning/common/lib/openplanner/op_ros_helpers/src/op_RosHelpers.cpp

```diff
@@ -238,73 +238,541 @@ void RosHelpers::CreateCircleMarker(const PlannerHNS::WayPoint& _center, const d
+
+    	if(lane_waypoints_array.lanes.at(i).waypoints.at(j).dtlane.dir == 1)
+    	{
+    		lane_waypoint_marker.color.r = 0.0;
+    		lane_waypoint_marker.color.g = 1.0;
+    		lane_waypoint_marker.color.b = 0.0;
+    		tmp_marker_array.markers.push_back(lane_waypoint_marker);
+    	}
+    	else if(lane_waypoints_array.lanes.at(i).waypoints.at(j).dtlane.dir == 2)
+    	{
+    		lane_waypoint_marker.color.r = 0.0;
+			lane_waypoint_marker.color.g = 0.0;
+			lane_waypoint_marker.color.b = 1.0;
+			tmp_marker_array.markers.push_back(lane_waypoint_marker);
+    	}
+    	else
+    	{
+    		lane_waypoint_marker.color.r = 1.0;
+			lane_waypoint_marker.color.g = 0.0;
+			lane_waypoint_marker.color.b = 1.0;
+			tmp_marker_array.markers.push_back(lane_waypoint_marker);
+    	}
+      count++;
+    }
+  }
+
+  markerArray.markers.insert(markerArray.markers.end(), tmp_marker_array.markers.begin(),
+										   tmp_marker_array.markers.end());
+}
+
+void RosHelpers::ConvertFromAutowareDetectedObjectToOpenPlannerDetectedObject(const autoware_msgs::DetectedObject& det_obj, PlannerHNS::DetectedObject& obj)
+{
+	obj.id = det_obj.id;
+	obj.label = det_obj.label;
+	obj.l = det_obj.dimensions.x;
+	obj.w = det_obj.dimensions.y;
+	obj.h = det_obj.dimensions.z;
+
+	obj.center.pos.x = det_obj.pose.position.x;
+	obj.center.pos.y = det_obj.pose.position.y;
+	obj.center.pos.z = det_obj.pose.position.z;
+	obj.center.pos.a = tf::getYaw(det_obj.pose.orientation);
+
+	obj.center.v = det_obj.velocity.linear.x;
+	obj.bVelocity = det_obj.velocity_reliable;
+	obj.bDirection = det_obj.pose_reliable;
+
+	PlannerHNS::GPSPoint p;
+	obj.contour.clear();
+
+	for(unsigned int j=0; j < det_obj.convex_hull.polygon.points.size(); j++)
+	{
+
+		p.x = det_obj.convex_hull.polygon.points.at(j).x;
+		p.y = det_obj.convex_hull.polygon.points.at(j).y;
+		p.z = det_obj.convex_hull.polygon.points.at(j).z;
+		obj.contour.push_back(p);
+	}
+
+	obj.predTrajectories.clear();
+
+	for(unsigned int j = 0 ; j < det_obj.candidate_trajectories.lanes.size(); j++)
+	{
+		std::vector<PlannerHNS::WayPoint> _traj;
+		PlannerHNS::RosHelpers::ConvertFromAutowareLaneToLocalLane(det_obj.candidate_trajectories.lanes.at(j), _traj);
+		obj.predTrajectories.push_back(_traj);
+	}
+}
+
+void RosHelpers::ConvertFromOpenPlannerDetectedObjectToAutowareDetectedObject(const PlannerHNS::DetectedObject& det_obj, autoware_msgs::DetectedObject& obj)
+{
+	obj.id = det_obj.id;
+	obj.label = det_obj.label;
+	obj.dimensions.x = det_obj.l;
+	obj.dimensions.y = det_obj.w;
+	obj.dimensions.z = det_obj.h;
+
+	obj.pose.position.x = det_obj.center.pos.x;
+	obj.pose.position.y = det_obj.center.pos.y;
+	obj.pose.position.z = det_obj.center.pos.z;
+	obj.pose.orientation = tf::createQuaternionMsgFromRollPitchYaw(0, 0, UtilityHNS::UtilityH::SplitPositiveAngle(det_obj.center.pos.a));
+
+	obj.velocity.linear.x = det_obj.center.v;
+	obj.velocity_reliable = det_obj.bVelocity;
+	obj.pose_reliable = det_obj.bDirection;
```

Comment:
direction is not pose

Reply:
I temporary use it this way between OpenPlanner nodes. I should add new variable to autoware_msg. will add a memo